Fix admin create event flow to create contract for invitee

albertchae opened the pull request:
## Summary of the problem
<!-- Why these changes are being made? What problem does it solve? Link any related issues to provide more details. -->

Fixes https://github.com/hackclub/hcb/issues/10738

When creating a new event in the admin flow, invited users would not have an associated contract created. This PR fixes that.

## Describe your changes

**Best reviewed by commits**

- extract a partial to ask for signee + cosigner + onboarding videos from regular invite flow and share it with admin flow
- currently this has is_signee default to true since that was the existing behavior for admins, we should revisit if that's what we want for regular invites too. It's slightly awkward to set the default due to the partial being shared now.

small QOL improvement to hide the cosigner field if the user is not going to be a signee, as it's not relevant. This will still get sent to the backend if a user filled it out but then flipped the `is_signee` flag off, but the backend correctly ignores it.

We might want to fold in contract creation into the OrganizerPositionInvite service somehow since they seem related.

albertchae committed:
Add stimulus controller to hide cosignee if not signing contract

albertchae committed:
Extract contract signee + cosigner question to a separate partial

This is in preparation to share this partial with the admin flow

albertchae committed:
Slight reformatting of existing code

- params list is really long so this makes it easier to see future diffs
- trailing if is after a long line, so this makes it clearer we only
  create the contract for certain users

albertchae committed:
Use shared partial for is_signee admin event create

- replace existing is_signee for admin event create with shared partial
- add call to OrganizerPosition::Contract in (Admin) EventService::Create to create contract
  We might want to move this call into OrganizerPositionInviteService::Create, but the
  tradeoff is we'll have to add cosigner_email and include_videos to the
  wrapper service

albertchae committed:
default contract signee status to true for admin event create

We might not want this for the regular organizer position invite flow
though, so need to double check how best to set this in a shared partial

garyhtou commented on app/views/organizer_position_invites/_admin_signee_check.html.erb:
```diff
@@ -0,0 +1,27 @@
+<div class="field field--select w-100"
```
Let's add a strict locals comment

https://blog.appsignal.com/2024/09/11/ruby-on-rails-7-1-partial-strict-locals-and-their-gotchas.html

garyhtou commented:
On the admin new organization page, let's default to contract signee true. On the org team page, deafult to false.

albertchae committed:
PR feedback

- simplify hiding logic to use hidden attribute
- add strict locals
- use default_selected local so we can have different defaults for where
  this partial is being used

albertchae committed:
Fix deprecation warning nil model to form_with

Explicitly passing a nil model to form_with will start throwing an
exception in Rails 8.0. In this case it was superfluous because we have
a url argument instead

DEPRECATION WARNING: Passing nil to the :model argument is deprecated and will raise in Rails 8.0

albertchae commented on app/javascript/controllers/form_hide_field_controller.js:
```diff
@@ -0,0 +1,23 @@
+    if (selectedValue === this.conditionValue) {
+      this.fieldTarget.style.display = "none"
+    } else {
+      this.fieldTarget.style.display = "block"
```
I instead decided to toggle `hidden` instead which seemed simpler than storing and restoring the display

garyhtou reviewed (approved):
we'll come back to fix those

albertchae commented:
> * reader / member / manager does not affect visibility of the contract tool
> 
>     * select dropdown has 3 options, one of them being to select despite being a binary option.
>       
>       * the select option shows the email box and should probably be removed.
>       * this same behavior occurs on the create event page

Addressing these in https://github.com/hackclub/hcb/pull/11011
